Source/Adobe.Target.Client/Service/TargetService.cs

```diff
@@ -10,20 +10,128 @@
  */
 namespace Adobe.Target.Client.Service
 {
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Threading.Tasks;
+    using Adobe.Target.Client.Model;
+    using Adobe.Target.Client.Util;
+    using Adobe.Target.Delivery.Api;
+    using Adobe.Target.Delivery.Client;
+    using Adobe.Target.Delivery.Model;
+
     /// <summary>
     /// Target Service
     /// </summary>
     public class TargetService
     {
-        private readonly ClientConfig clientConfig;
+        private readonly TargetClientConfig clientConfig;
+        private DeliveryApi deliveryApi;
+        private string stickyLocationHint;
+        private volatile string stickyBaseUrl;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="TargetService"/> class.
         /// </summary>
-        /// <param name="clientConfig">Target Client Config <see cref="ClientConfig"/></param>
-        public TargetService(ClientConfig clientConfig)
+        /// <param name="clientConfig">Target Client Config <see cref="TargetClientConfig"/></param>
+        public TargetService(TargetClientConfig clientConfig)
         {
             this.clientConfig = clientConfig;
+            this.stickyBaseUrl = this.clientConfig.DefaultUrl;
```

Comment:
Could we initialize `deliveryApi` here once without using a `void` method.

Reply:
We need the `setDeliveryApi` method as we need to re-create it each time baseUrl changes.